src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTokenCheckTest.java

```diff
@@ -79,20 +83,29 @@ public void testCommentContentToken()
                 createModuleConfig(IllegalTokenCheck.class);
         checkConfig.addAttribute("tokens", "COMMENT_CONTENT");
 
+        final String path = getPath("InputIllegalTokens.java");
+        final String lineSeparator;
+        if (new FileText(new File(path), StandardCharsets.UTF_8.name())
+                .getFullText().chars().anyMatch(character -> character == '\r')) {
+            lineSeparator = "\r\n";
+        }
+        else {
+            lineSeparator = "\n";
+        }
         final String[] expected = {
             "3:3: " + getCheckMessage(MSG_KEY,
                         JavadocUtil.escapeAllControlChars(
-                            "*" + System.lineSeparator()
+                            "*" + lineSeparator
```

Comment:
Well, it for one still depends on whether you will set the attributes, so that LF is enforced on checkout or whether you set the attributes so that the user can still decide how he wants the files checked out, but they get normalized to LF on checkin automatically.
I thought we agreed on the latter but I might be mistaken.

Also I'm not sure about this exact location, there are also some tests where output from checkstyle is verified that is generated with the system default line separator and then needs to be normalized for the tests. I think this is not such a case, but like it is currently we still support having the files checked out with CRLF or LF and the tests just work.

Reply:
> so that LF is enforced on checkout 

yes, we need to enforce this by ".gitattributes" file. So user should not bother about any action on clone. Git will do all for user, sources will be "\n", build will work fine out of the box.
If user do extra override in configuration ... he do know what he is doing, it mean he is advanced  user, he will resolve all problems. 

> I thought we agreed on the latter but I might be mistaken.

no, at least as for me.
All other looks like took neutral position. 

> but like it is currently we still support having the files checked out with CRLF or LF and the tests just work.

As windows text editors / IDEs do not have problems with linux line ending. I would rather enforce single lineend - "\n" for all files except for "*.bat,*.ps"(first one is almost in repo - #6750).  
So code state will be more predictable on all OSs.
Checkstyle is already over-ruled with thousands of CI rule/checks/validations .... so contributor is already accustomed that code need to be consistent/exact/pristine even is very-very-very tiny aspect of it.

Reply:
> > so that LF is enforced on checkout
> 
> yes, we need to enforce this by ".gitattributes" file. So user should not bother about any action on clone. Git will do all for user, sources will be "\n", build will work fine out of the box.
> If user do extra override in configuration ... he do know what he is doing, it mean he is advanced user, he will resolve all problems.

Well, there wouldn't be any problem if the test just support it.
And he cannot resolve it in that case short of changing the test that could already support it without a problem.
And also that he has such a setting doesn't mean that he is andvanced, just that he can follow probably stupid advices from StackOverflow or similar.
But if you really want it that way, you need to do the `.gitattributes` change first I think.

> > I thought we agreed on the latter but I might be mistaken.
> 
> no, at least as for me.
> All other looks like took neutral position.

Ah, sorry, remembered wrongly.
Actually we didn't agree yet.
You asked whether I agree and I asked whether I can say "no", but we didn't continue on that part.
Imho it should still be sufficient to force the normalization on commit and let the user decide what lineending he wants to have locally.

> As windows text editors / IDEs do not have problems with linux line ending. I would rather enforce single lineend - "\n" for all files except for "_.bat,_.ps"(first one is almost in repo - #6750).

I'd rather say "many", not the implied "all".
But yeah, if you really want it like that, fine.

> Checkstyle is already over-ruled with thousands of CI rule/checks/validations .... so contributor is already accustomed that code need to be consistent/exact/pristine even is very-very-very tiny aspect of it.

contributors, yes
But this does not only hit contributors but also users that just want to build the sources. :-)

Reply:
> But if you really want it that way, you need to do the .gitattributes change first I think.

if we make it first, we will not accept your update for tests. 

> I can say "no", but we didn't continue on that part.

please look https://github.com/checkstyle/checkstyle/pull/6559#issuecomment-502420662 

> but also users that just want to build the sources. :-)

if they are interesting in build, they can skip tests.